language/bytecode-verifier/src/dependencies.rs

```diff
@@ -338,3 +338,46 @@ impl<'a> DependencyChecker<'a> {
         }
     }
 }
+
+pub struct CyclicModuleDependencyChecker {}
+
+impl CyclicModuleDependencyChecker {
+    pub fn verify_module<F>(module: &CompiledModule, dep_loader: F) -> VMResult<()>
+    where
+        F: Fn(&ModuleId) -> Vec<ModuleId>,
+    {
+        Self::verify_module_impl(module, dep_loader)
+            .map_err(|e| e.finish(Location::Module(module.self_id())))
+    }
+
+    pub fn verify_module_impl<F>(module: &CompiledModule, dep_loader: F) -> PartialVMResult<()>
+    where
+        F: Fn(&ModuleId) -> Vec<ModuleId>,
+    {
+        // A standard DFS algorithm that follows the dependency chain from the `cursor_module_id`
+        // to see whether the `target_module_id` is in its dependencies.
+        fn is_reachable_via_dependencies<F>(
+            target_module_id: &ModuleId,
+            cursor_module_id: &ModuleId,
+            dep_loader: &F,
```

Comment:
I think you can replace  `&F` with `F`. 

&dep_loader should implement the desired trait here

Reply:
I thought so, but the compiler does not seem to be happy with it:
`error[E0275]: overflow evaluating the requirement [closure@language/move-vm/runtime/src/loader.rs:649:62: 653:10]: Fn<(&ModuleId,)>`

Reply:
Thats a new one...